src/wrappers/gdalDemProcessing.js

```diff
@@ -0,0 +1,145 @@
+        // TODO: Make this unnecessary by switching to comlink or similar (#49)
+        const mode = packedArgs[0];
+        const colors = packedArgs[1];
+        const args = packedArgs.slice(2);
+
+        if (!mode || !DEMProcessingModes.hasOwnProperty(mode)) {
+            throw new Error(`mode must be one of {Object.keys(DEMProcessingModes)}`);
+        } else if (mode === DEMProcessingModes['color-relief'] && !colors) {
+            throw new Error(
+                'A color definition array must be provided if `mode` is "color-relief"'
+            );
+        } else if (mode !== DEMProcessingModes['color-relief'] && colors && colors.length > 0) {
+            throw new Error(
+                'A color definition array should not be provided if `mode` is not "color-relief"'
+            );
+        }
+
+        // If mode is hillshade, we need to create a color file path
+        let colorFilePath = null;
+
+        if (mode === DEMProcessingModes['color-relief']) {
+            colorFilePath = rootPath + randomKey() + '.txt';
+
+            FS.writeFile(colorFilePath, colors.join('\n'));
+        }
+        let params = new ParamParser(args);
+
+        params.allocate();
+
+        // argPtrsArrayPtr is now the address of the start of the list of
+        // pointers in Emscripten heap space. Each pointer identifies the address of the start of a
+        // parameter string, also stored in heap space. This is the direct equivalent of a char **,
+        // which is what GDALDEMProcessingOptionsNew requires.
+        const demOptionsPtr = Module.ccall(
+            'GDALDEMProcessingOptionsNew',
+            'number',
+            ['number', 'number'],
+            [params.argPtrsArrayPtr, null]
+        );
+
+        // Validate that the options were correct
+        const optionsErrType = errorHandling.CPLGetLastErrorType();
+
+        if (
+            optionsErrType === errorHandling.CPLErr.CEFailure ||
+            optionsErrType === errorHandling.CPLErr.CEFatal
+        ) {
+            if (colorFilePath) {
+                FS.unlink(colorFilePath);
+            }
+            params.deallocate();
+            const message = errorHandling.CPLGetLastErrorMsg();
+
+            throw new Error('Error in GDALDEMProcessing: ' + message);
+        }
+
+        // Now that we have our options, we need to make a file location to hold the output.
+        let directory = rootPath + randomKey();
+
+        FS.mkdir(directory);
+        // This makes it easier to remove later because we can just unmount rather than recursing
+        // through the whole directory structure.
+        FS.mount(MEMFS, {}, directory);
+        let filename = randomKey(8) + '.' + guessFileExtension(args);
+
+        let filePath = directory + '/' + filename;
+
+        // And then we can kick off the actual processing.
+        // TODO: The last parameter is an int* that can be used to detect certain kinds of errors,
```

Comment:
Should this TODO be an issue to investigate?

Reply:
Good call, I think this should more just not be a `TODO` at all -- I've been able to get pretty good error messages out of the `CPLGetLastError*` methods, so it hasn't felt necessary to dig into what this does so far.